### PR TITLE
Add reusable span tracer module

### PR DESCRIPTION
**Pull request by maciej-szlosarczyk**
This adds a reusable tracer module, which in example of the custom tracer we use in stories, shortens it from 90 to 55 lines of code. Also in use here in absinthe and liveview tracers.

Signed-off-by: Maciej Szlosarczyk <maciej@mindvalley.com>

### CODE REVIEW

File: test/mv_opentelemetry/span_tracer_test.exs
Comment (feliperenan) [inline test/mv_opentelemetry/span_tracer_test.exs:37]: Could you use [assert_raise/2](https://hexdocs.pm/ex_unit/1.12/ExUnit.Assertions.html#assert_raise/2) instead? 🤔

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/9#pullrequestreview-782101774" title="Approved">feliperenan</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **maciej-szlosarczyk** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/mindvalley/mv-opentelemetry/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [e1c2598](https://github.com/mindvalley/mv-opentelemetry/commits/e1c2598a867aae4768fb20720d396915c817c664) Add feliperenan to approvers

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: test/mv_opentelemetry/span_tracer_test.exs
Comment (maciej-szlosarczyk) [reply test/mv_opentelemetry/span_tracer_test.exs:37]: I don't think so - I'm not really interested in the error itself, but I'm looking at what happens after that. I could probably use a task to fire it off, but this way it seems more readable.

File: lib/mv_opentelemetry.ex
Comment (maciej-szlosarczyk) [inline lib/mv_opentelemetry.ex:92]: These two are slightly breaking changes, but nobody is using them AFAIK so it should be fine.

File: lib/mv_opentelemetry/span_tracer.ex
Comment (Vollmond) [inline lib/mv_opentelemetry/span_tracer.ex:7]: And at maximum?

Comment (maciej-szlosarczyk) [reply lib/mv_opentelemetry/span_tracer.ex:7]: Nah, you can also override the `c:register_tracer/1`. I think at least would be a better wording, will update.

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **maciej-szlosarczyk** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
@maciej-szlosarczyk: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Review by Vollmond — APPROVED** [review]

**Comment by sauron-droid** [comment]
LGTM label has been added.  <details>Git tree hash: 1e6e7a949d73b2dcc895ac2671707ec59efc4c7d</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/9#pullrequestreview-796205291" title="Approved">Vollmond</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)~~ [Vollmond]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
